forgerock-auth/src/main/java/org/forgerock/android/auth/AppAuthConfigurer.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2020 ForgeRock. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+package org.forgerock.android.auth;
+
+import android.net.Uri;
+
+import androidx.annotation.NonNull;
+import androidx.browser.customtabs.CustomTabsIntent;
+import androidx.core.util.Consumer;
+
+import net.openid.appauth.AppAuthConfiguration;
+import net.openid.appauth.AuthorizationRequest;
+import net.openid.appauth.AuthorizationServiceConfiguration;
+
+import java.net.MalformedURLException;
+
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * AppAuth Integration, https://github.com/openid/AppAuth-Android
+ * this class provides AppAuth integration and customization
+ */
+@RequiredArgsConstructor
+public class AppAuthConfigurer {
+
+    private final FRUser.Browser parent;
+
+    @Getter(AccessLevel.PACKAGE)
+    private Consumer<AuthorizationRequest.Builder> authorizationRequestBuilder = builder -> {
+    };
+    @Getter(AccessLevel.PACKAGE)
+    private androidx.core.util.Consumer<AppAuthConfiguration.Builder> appAuthConfigurationBuilder = builder -> {
+    };
+    @Getter(AccessLevel.PACKAGE)
+    private Consumer<CustomTabsIntent.Builder> customTabsIntentBuilder = builder -> {
+    };
+    @Getter(AccessLevel.PACKAGE)
+    private Supplier<AuthorizationServiceConfiguration> authorizationServiceConfigurationSupplier = () -> {
+        OAuth2Client oAuth2Client = Config.getInstance().getOAuth2Client();
+        try {
+            return new AuthorizationServiceConfiguration(
+                    Uri.parse(oAuth2Client.getAuthorizeUrl().toString()),
+                    Uri.parse(oAuth2Client.getTokenUrl().toString()));
+        } catch (MalformedURLException e) {
+            throw new RuntimeException(e);
+        }
+    };
+
+    /**
+     * Override the default OAuth2 endpoint defined under the String.xml
+     * forgerock_url
+     */
+    public AppAuthConfigurer authorizationServiceConfiguration(@NonNull Supplier<AuthorizationServiceConfiguration> authorizationServiceConfiguration) {
+        this.authorizationServiceConfigurationSupplier = authorizationServiceConfiguration;
+        return this;
+    }
+
+    /**
+     * Override the {@link AuthorizationRequest} that prepared by the SDK.
```

Comment:
Do you mean 'that was prepared by the SDK.'?

Reply:
updated